controller/controller.go

```diff
@@ -220,169 +175,120 @@ func (ac *ALBController) ingressToDelete(newList ALBIngressesT) ALBIngressesT {
-		if !ok {
-			log.Infof("The LoadBalancer %s does not have a Hostname tag, can't import", "controller", *loadBalancer.LoadBalancerName)
-			continue
-		}
-
-		ingressID := namespace + "-" + ingressName
+	var wg sync.WaitGroup
+	wg.Add(len(loadBalancers))
 
-		var rs *alb.ResourceRecordSet
+	for _, loadBalancer := range loadBalancers {
+		go func(wg *sync.WaitGroup, loadBalancer *elbv2.LoadBalancer) {
+			defer wg.Done()
 
-		if !ac.disableRoute53 {
-			zone, err := awsutil.Route53svc.GetZoneID(&hostname)
-			if err != nil {
-				log.Infof("Failed to resolve %s zoneID. Returned error %s", "controller", hostname, err.Error())
-				continue
+			albIngress, ok := NewALBIngressFromLoadBalancer(loadBalancer, *ac.ClusterName)
+			if !ok {
+				return
 			}
 
-			log.Infof("Fetching resource recordset for %s/%s %s", "controller", namespace, ingressName, hostname)
-			resourceRecordSet, err := awsutil.Route53svc.DescribeResourceRecordSets(zone.Id,
-				&hostname)
-			if err != nil {
-				log.Errorf("Failed to find %s in AWS Route53", ingressID, hostname)
+			if i := ac.ALBIngresses.find(albIngress); i >= 0 {
+				albIngress = ac.ALBIngresses[i]
+				albIngress.LoadBalancers = append(albIngress.LoadBalancers, albIngress.LoadBalancers[0])
+			} else {
+				ac.ALBIngresses = append(ac.ALBIngresses, albIngress)
 			}
+		}(&wg, loadBalancer)
+	}
+	wg.Wait()
 
-			rs = &alb.ResourceRecordSet{
-				IngressID: &ingressID,
-				ZoneID:    zone.Id,
-				CurrentResourceRecordSet: resourceRecordSet,
-			}
-		} else {
-			log.Warnf("Route53 disabled", ingressID)
-			rs = nil
-		}
+	log.Infof("Assembled %d ingresses from existing AWS resources", "controller", len(ac.ALBIngresses))
+}
 
-		lb := &alb.LoadBalancer{
-			ID:                  loadBalancer.LoadBalancerName,
-			IngressID:           &ingressID,
-			Hostname:            aws.String(hostname),
-			CurrentLoadBalancer: loadBalancer,
-			ResourceRecordSet:   rs,
-			CurrentTags:         tags,
-		}
+// StateHandler returns the JSON state of the controller
+func (ac *ALBController) StateHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(ac.ALBIngresses)
+}
 
-		targetGroups, err := awsutil.ALBsvc.DescribeTargetGroups(loadBalancer.LoadBalancerArn)
-		if err != nil {
-			glog.Fatal(err)
-		}
+// ConfigureFlags
+func (ac *ALBController) ConfigureFlags(pf *pflag.FlagSet) {
+	pf.StringVar(ac.ClusterName, "cluster-name", "", "The name of the cluster, used for naming AWS resources")
+	pf.BoolVar(&ac.disableRoute53, "disable-route53", false, "Disable Route 53 management")
+}
 
-		for _, targetGroup := range targetGroups {
-			tags, err := awsutil.ALBsvc.DescribeTags(targetGroup.TargetGroupArn)
-			if err != nil {
-				glog.Fatal(err)
-			}
+// OverrideFlags configures optional override flags for the ingress controller
+func (ac *ALBController) OverrideFlags(flags *pflag.FlagSet) {
+	flags.Set("update-status-on-shutdown", "false")
```

Comment:
We want to get this into master, without this the address will get removed on shutdown even though we dont destroy the AWS resources